django/db/models/sql/compiler.py

```diff
@@ -340,7 +340,8 @@ def quote_name_unless_alias(self, name):
         if name in self.quote_cache:
             return self.quote_cache[name]
         if ((name in self.query.alias_map and name not in self.query.table_map) or
-                name in self.query.extra_select or name in self.query.external_aliases):
+                name in self.query.extra_select or (
+                    name in self.query.external_aliases and name not in self.query.table_map)):
```

Comment:
Ahh we were so close, we tried removing `or name in self.query.external_aliases` but this triggered more failures. This all makes sense now.